### ISSUE TITLE
Validate if the diskOffering field is modified during Upgrade for Cloudstack 

### ISSUE THREAD

**Issue by rahulbabu95** [aws/eks-anywhere#5319]
**What would you like to be added**:
Add a pre-flight validation to check if the diskOffering field is added/modified during the upgrade operation and fail the upgrade in pre-flight if there is a change in value.

**Why is this needed**:
Currently, when we a add a diskOffering field to the cluster spec which did not have any disk mounts previously during the upgrade, the upgrade fails as the diskSetup.partitions field is immutable. But the failure only happens after the bootstrap cluster is created and all the management objects are moved to the Kind cluster. 

```
2023-03-21T00:07:04.183Z	V5	Error happened during retry	{"error": "executing apply: Warning: resource kubeadmcontrolplanes/rahul-mgmt is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.\nThe KubeadmControlPlane \"rahul-mgmt\" is invalid: \n* spec.kubeadmConfigSpec.diskSetup.filesystems: Forbidden: cannot be modified\n* spec.kubeadmConfigSpec.diskSetup.partitions: Forbidden: cannot be modified\n* spec.kubeadmConfigSpec.mounts: Forbidden: cannot be modified\n", "retries": 30}
2023-03-21T00:07:04.183Z	V5	Execution aborted by retry policy
```

We can add a pre-flight to check for changes in diskOffering field and fail early.

**Comment by chrisdoherty4** [comment]
CAPI v1.4.x starts to allow the fields referenced in the error to be mutated. I think the best course of action is to publish a note with our next release and highlight the issue will be fixed when we upgrade to CAPI v1.4.

Change available in the v1.4.0 release: https://github.com/kubernetes-sigs/cluster-api/blob/v1.4.0/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go#L172-L175

**Comment by ddjjia** [comment]
will test it again once we consume 1.4
